haiku-apps/begui/patches/begui-1.4.2.patchset

```diff
@@ -0,0 +1,149 @@
+From b0167363cf1f58f5a25732d064828eea154e2160 Mon Sep 17 00:00:00 2001
+From: Arnav Bhatt <arnavbhatt288@gmail.com>
+Date: Sun, 18 Nov 2018 20:11:52 +0000
+Subject: Fixing BeGUI
+
+
+diff --git a/BeGUIlib/basicColorWindow.cpp b/BeGUIlib/basicColorWindow.cpp
+index cad4a5c..3fd7a91 100644
+--- a/BeGUIlib/basicColorWindow.cpp
++++ b/BeGUIlib/basicColorWindow.cpp
+@@ -264,6 +264,7 @@ BasicColorWindow	::	MessageReceived(	BMessage * paramMessage)
+     		rgb_color * pRGBColor;
+     		rgb_color color;  		
+     		int32 numRGBBytes = sizeof(struct rgb_color);
++    #ifdef SOMETHING	
+ 	        if (	paramMessage->FindData(	"RGBColor",
+ 		       									B_RGB_COLOR_TYPE, 
+ 												(const void**)&pRGBColor, 
+@@ -304,6 +305,7 @@ BasicColorWindow	::	MessageReceived(	BMessage * paramMessage)
+ 					}
+ 				}
+ 			}
++	#endif		
+ 		break;
+ 		}
+ 		default:
+diff --git a/BeGUIlib/makefile.lib b/BeGUIlib/makefile.lib
+index 27e5805..71164c2 100644
+--- a/BeGUIlib/makefile.lib
++++ b/BeGUIlib/makefile.lib
+@@ -124,7 +124,7 @@ RSRCS =
+ #	- 	if your library does not follow the standard library naming scheme,
+ #		you need to specify the path to the library and it's name.
+ #		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
+-LIBS = prefs be root tracker
++LIBS = prefs be root tracker $(STDCPPLIBS)
+ 
+ #	Specify additional paths to directories following the standard libXXX.so
+ #	or libXXX.a naming scheme. You can specify full paths or paths relative
+diff --git a/BeGUIlib/myRadioView.cpp b/BeGUIlib/myRadioView.cpp
+index 95a045a..1f428e6 100644
+--- a/BeGUIlib/myRadioView.cpp
++++ b/BeGUIlib/myRadioView.cpp
+@@ -11,6 +11,7 @@
+ #include "Preferences.h"
+ #include "layoutMatrix.h"
+ #include "myPreferences.h"
++#include <memory>
+ 
+ MyRadioView	::	~MyRadioView() 
+ {
+@@ -67,7 +68,7 @@ MyRadioView	::	Initialize(	void)
+ 	SetFont(mpFont);
+ 	try
+ 	{
+-		auto_ptr<BList> apButtonList(new BList(/*numItems*/));
++		std::auto_ptr<BList> apButtonList(new BList(/*numItems*/));
+ 		BList * pButtonList = apButtonList.get();
+ 		BRadioButton * holdBRB;
+ 		bool oneSelected = false;
+@@ -188,4 +189,4 @@ RadioViewData	::	RadioViewData(	uint32 paramMessageValue,
+ RadioViewData	::	~RadioViewData()
+ {
+ 	//leaving blank
+-}
+\ No newline at end of file
++}
+diff --git a/BeGUIlib/specificColorWindow.cpp b/BeGUIlib/specificColorWindow.cpp
+index 88f3c34..498b331 100644
+--- a/BeGUIlib/specificColorWindow.cpp
++++ b/BeGUIlib/specificColorWindow.cpp
+@@ -676,6 +676,7 @@ SpecificColorWindow	::	MessageReceived(	BMessage * paramMessage)
+     		rgb_color * pRGBColor;
+     		rgb_color color;   		
+     		int32 numRGBBytes = sizeof(struct rgb_color);
++    	#ifdef	
```

Comment:
You should ifdef SOMETHING here, same as the one before, but it should fit some better description/pointer on why it's needed?

Reply:
On that note, builds ok without this #ifdef's ?